### ISSUE TITLE
[Docs]: Confusing messaging about v7_partialHydration and HydrateFallback

### ISSUE THREAD

**Issue by jdufresne** [remix-run/react-router#12249]
### Describe what's incorrect/missing in the documentation

When upgrading from 6.27.0 to 6.28.0, I started to see the console error:

> ⚠️ React Router Future Flag Warning: `RouterProvider` hydration behavior is changing in v7. You can use the `v7_partialHydration` future flag to opt-in early. For more information, see https://reactrouter.com/v6/upgrading/future#v7_partialhydration.

I followed the link which says:

> This allows SSR frameworks to provide only partial hydration data. It's unlikely you need to worry about this, just turn the flag on.

I am not using a SSR framework, so I followed the advice of "just turn the flag on". After doing so, I started to receive a new error:

> No `HydrateFallback` element provided to render during initial hydration

This error does not include a helpful link, so I searched the docs and found:

https://reactrouter.com/en/main/route/hydrate-fallback-element

> If you are using Server-Side Rendering and you are leveraging partial hydration, then you can specify an Element/Component to render for non-hydrated routes during the initial hydration of the application.

As I'm not using SSR, I'm confused as to why I need to specify this and what I should specify it as. Without knowing better, I specified this as `React.Fragment`, but it is unclear to me if this is an appropriate value.

My suggestion would be to:

1. Better explain why I should be specifying this when I'm not using SSR.
2. Guide me towards what a useful value may look like and how this affects my application.

**Comment by timdorr** [comment]
Duplicate of #12250 

The main issue is that we have too many warnings in this version, regardless of some being confusingly configured.

**Comment by Mori-Young** [comment]
I'm very sorry to bother you. I'm not using SSR, but the console still warns: No `HydrateFallback` element provided to render during initial hydration Error Component Stack

How should I solve this problem? @timdorr @jasperfirecai2

**Comment by jasperfirecai2** [comment]
you can copy the example code from the docs to make the warning go away
```jsx
export function HydrateFallback() {
  return <p>Loading Game...</p>;
}
```

**Comment by Mori-Young** [comment]
But I used react router as a library, not a framework. I didn't know where to use `HydrateFallback`.
Finally I added `hydrateFallbackElement` to one of the routes in `createBrowserRouter`, and the warning went away.

I just came into contact with react router, and the learning process made me feel incredible. I don't understand why it should be used as a framework, and my understanding of the documentation is limited.
